src/UNL/Peoplefinder/Record/Avatar.php

```diff
@@ -155,17 +161,26 @@ protected function generatePersonUrl($options)
         $planetRedUid = $this->record->getProfileUid();
         $profileIconUrl = UNL_Peoplefinder_Record::PLANETRED_BASE_URL . 'icon/' . 'unl_' . $planetRedUid . '/' . $size . '/';
 
-        $request = new HTTP_Request2($profileIconUrl, HTTP_Request2::METHOD_HEAD, [
-            'adapter' => 'HTTP_Request2_Adapter_Curl',
-            'follow_redirects' => true,
-            'strict_redirects' => true,
+        $effectiveUrl = $profileIconUrl;
+        $onRedirect = function(
+            RequestInterface $request,
+            ResponseInterface $response,
+            UriInterface $uri
+        ) use (&$effectiveUrl) {
+            $effectiveUrl = (string) $uri;
```

Comment:
This is clever, but I wonder if `$effectiveUrl = $response->getEffectiveUrl()` would be easier to read